client/db/bolt/upgrades_test.go

```diff
@@ -0,0 +1,93 @@
+// This code is available on the terms of the project LICENSE.md file,
+// also available online at https://blueoakcouncil.org/license/1.0.0.
+
+package bolt
+
+import (
+	"compress/gzip"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"go.etcd.io/bbolt"
+)
+
+var dbUpgradeTests = [...]struct {
+	name     string
+	verify   func(*testing.T, *bbolt.DB)
+	filename string // in testdata directory
+}{
+	{"upgradeFromV0", verifyV1Upgrade, "v0.db.gz"},
+}
+
+func TestUpgrades(t *testing.T) {
+	t.Parallel()
+
+	d, err := ioutil.TempDir("", "dcrdex_test_upgrades")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for _, tc := range dbUpgradeTests {
+		t.Run(tc.name, func(t *testing.T) {
+			testFile, err := os.Open(filepath.Join("testdata", tc.filename))
+			if err != nil {
+				t.Fatal(err)
+			}
+			defer testFile.Close()
+			r, err := gzip.NewReader(testFile)
+			if err != nil {
+				t.Fatal(err)
+			}
+			dbPath := filepath.Join(d, tc.name+".db")
+			fi, err := os.Create(dbPath)
+			if err != nil {
+				t.Fatal(err)
+			}
+			_, err = io.Copy(fi, r)
+			fi.Close()
+			if err != nil {
+				t.Fatal(err)
+			}
+			db, err := bbolt.Open(dbPath, 0600,
+				&bbolt.Options{Timeout: 1 * time.Second})
+			if err != nil {
+				t.Fatal(err)
+			}
+			defer db.Close()
+			err = upgradeDB(db)
```

Comment:
What is stopping `upgradeDB` from going to the final version, say 2, thus failing `verifyV1Upgrade` in `dbUpgradeTests[0].verify`?  Seems you can't test `upgradeDB` in a loop like this but rather the individual upgrade functions.